### PR TITLE
Refactor basic b

### PR DESCRIPTION
**Pull request by GabrielLyonB**
## Objective
- Include new modifications from las PR #1.
- Change database from sqlite3 to postgres.
- Include `counter_cache` in `Node` model.
- Move `Queable` concern to new folder.
- Move some of the concern methods to the model `Queue`.
- Include trait in to `Queue` factory.
- Add new tests.

### CODE REVIEW

File: queue_it.gemspec [hunk 1]
```diff
@@ -20,11 +20,13 @@ Gem::Specification.new do |s|
   s.test_files = Dir["spec/**/*"]
 
   s.add_dependency "rails", ">= 6.0"
+  s.add_development_dependency "annotate", "~> 3.0"
```
Comment (ldlsegovia) [inline queue_it.gemspec:23]: buena, debería agregarle al generador

**Review by ldlsegovia — APPROVED** [review]

File: queue_it.gemspec (hunk 1 above)
Comment (GabrielLyonB) [reply queue_it.gemspec:23]: El generador de `annotate`? agregue las configuraciones si no me equivoco

Comment (ldlsegovia) [reply queue_it.gemspec:23]: A https://github.com/platanus/gemaker digo

Comment (GabrielLyonB) [reply queue_it.gemspec:23]: Ah buena! sí

**Pull request merged**